### PR TITLE
gnrc_netdev2: fix documentation

### PR DESCRIPTION
**Pull request by authmillenon**
Fixes some documentation issues in `gnrc_netdev2`. (The forward declaration was throwing an error in doxygen but is not really needed)

### CODE REVIEW

File: sys/include/net/gnrc/gnrc_netdev2.h [hunk 1]
```diff
@@ -9,14 +9,14 @@
 /**
  * @defgroup  net_gnrc_netdev2   Adaption layer for GNRC on top of Netdev2
  * @ingroup   net_gnrc
- * @brief     Provides the glue code for @ref gnrc on top of @ref drivers_netdev_netdev2
+ * @brief     Provides the glue code for @ref net_gnrc on top of @ref drivers_netdev_netdev2
  * @{
  *
  * @file
- * @brief     netdev2 gnrc glue code interface
+ * @brief     netdev2-GNRC glue code interface
```
Comment (OlegHahm) [inline sys/include/net/gnrc/gnrc_netdev2.h]: So, you propose `GNRC` (with capital letters) as the canonical form? (Just asking.)

File: sys/include/net/gnrc/gnrc_netdev2.h [hunk 2]
```diff
@@ -37,26 +37,27 @@
...
+ * @brief Structure holding GNRC netdev2 adapter state
  *
  * This structure is supposed to hold any state parameters needed
- * to use a netdev2 device from gnrc.
+ * to use a netdev2 device from GNRC.
  *
  * It can be extended
  */
-struct gnrc_netdev2 {
+typedef struct gnrc_netdev2 {
```
Comment (OlegHahm) [inline sys/include/net/gnrc/gnrc_netdev2.h]: That's not documentation and should be discussed separately IMO.

File: sys/include/net/gnrc/gnrc_netdev2.h (hunk 1 above)
Comment (authmillenon) [inline sys/include/net/gnrc/gnrc_netdev2.h]: From my standpoint, yes. But in general there are mixed occurrences in this file, so I unified it to the form more often used.

File: sys/include/net/gnrc/gnrc_netdev2.h (hunk 2 above)
Comment (authmillenon) [inline sys/include/net/gnrc/gnrc_netdev2.h]: It's documentation related, since doxygen was complaining about the undocumented

```C
typedef struct gnrc_netdev2 gnrc_netdev_t;
```

line. As shown it is not necessary to do this, however.

This is why I put it in a separate commit.

File: sys/include/net/gnrc/gnrc_netdev2.h (hunk 1 above)
Comment (OlegHahm) [inline sys/include/net/gnrc/gnrc_netdev2.h]: Ok, I think it's good to have a canonical spelling.

File: sys/include/net/gnrc/gnrc_netdev2.h (hunk 2 above)
Comment (OlegHahm) [inline sys/include/net/gnrc/gnrc_netdev2.h]: Ok, thanks for clarification.

**Comment by authmillenon** [comment]
Rebased to current master

**Comment by OlegHahm** [comment]
ACK

**Comment by authmillenon** [comment]
And go
